src/components/Matrix.vue

```diff
@@ -2,20 +2,36 @@
     <div class="matrix">
         <h2>Code-Matrix</h2>
         <div class="matrix-row" v-for="i in size" :key="i">
-            <div class="matrix-column" v-for="j in size" :key="j">
-                {{matrix[(i - 1)*5 + (j - 1)]}}
+            <div class="matrix-column" v-for="j in size" :key="j" v-on:click="select(i, j)">
+                {{matrix[(j - 1)*size + (i - 1)]}}
             </div>
         </div>
     </div>
 </template>
 
 <script lang="ts">
-    import {defineComponent} from "vue";
+    import { Game } from "@/game/Game";
+    import {defineComponent, inject, computed, Ref} from "vue";
     
     export default defineComponent({
-        props: {
-            size: {type: Number, required: true},
-            matrix: {type: Array, required: true},
+        setup() {
+            const game = inject("game") as Ref<Game>;
```

Comment:
does the game automatically update if you inject a different one? I would have expected the game to be a prop (or to be acquired via some kind of game manager.

Reply:
Yes it gets updated. Don't need to squash.